R/stacking.R

```diff
@@ -0,0 +1,1267 @@
+    omittable.names <- omittableNames(names(stacked.data.set),
+                                      include.original.case.variable,
+                                      include.observation.variable)
+    warning.if.not.found <- parse.status[parse.status != "bad format"] ==
+                                "variable(s) not found"
+
+    parsed.variables.to.omit.2 <- parseVariablesToOmit(variables.to.omit,
+                                                       omittable.names,
+                                                       warning.if.not.found)
+    omitted.stacked.variables <-  omittedStackedVariables(parsed.variables.to.omit.2,
+                                                          stacked.data.set)
+    stacked.data.set <- omitVariablesFromDataSet(stacked.data.set,
+                                                 parsed.variables.to.omit.2)
+
+    stacked.data.set.name <- cleanStackedDataSetName(stacked.data.set.name,
+                                                     input.data.set.name)
+    if (write.data.set)
+        writeDataSet(stacked.data.set, stacked.data.set.name)
+
+    stacked.data.set.metadata <- metadataFromStackedDataSet(stacked.data.set,
+                                                            stacked.data.set.name)
+
+
+    result <- list()
+    result$stacked.data.set.metadata <- stacked.data.set.metadata
+    result$unstackable.names <- attr(stacking.groups, "unstackable.names")
+    result$omitted.variables <- c(parsed.variables.to.omit, parsed.variables.to.omit.2)
+    result$omitted.stacked.variables <- omitted.stacked.variables
+    result$common.labels <- common.labels
+    result$is.saved.to.cloud <- write.data.set && canAccessDisplayrCloudDrive()
+
+    if (include.stacked.data.set.in.output)
+        result$stacked.data.set <- stacked.data.set
+
+    class(result) <- "StackedData"
+    result
+}
+
+# TODO: check for places where drop = FALSE is needed!
+
+readDataSets <- function(data.set.names, min.data.sets = 1)
+{
+    if (length(data.set.names) < min.data.sets)
+        stop("At least ", min.data.sets, " data set(s) are required.")
+
+    if (canAccessDisplayrCloudDrive())
+        readDataSetsFromDisplayrCloudDrive(data.set.names)
+    else
+        readLocalDataSets(data.set.names)
+}
+
+canAccessDisplayrCloudDrive <- function()
+{
+    company.secret <- get0("companySecret")
+    !is.null(company.secret) && company.secret != "UNKNOWN"
+}
+
+#' @importFrom haven read_sav
+readLocalDataSets <- function(data.set.paths)
+{
+    result <- lapply(data.set.paths, function(path) {
+        read_sav(path)
+    })
+    names(result) <- basename(data.set.paths)
+    result
+}
+
+#' @importFrom flipAPI QFileExists QLoadData
+readDataSetsFromDisplayrCloudDrive <- function(data.set.names)
+{
+    result <- lapply(data.set.names, function(nm) {
+        if (!QFileExists(nm))
+            stop("The data file '", nm, "' does not exist in the Display ",
```

Comment:
`Display` should be `Displayr`

Reply:
Fixed in 9802abe5634b4783532b0460f8c0491eae63646e